source/aws_imds_client.c

```diff
@@ -506,16 +499,30 @@ static void s_client_on_token_response(struct imds_user_data *user_data) {
      * token in following requests.
      */
     if (user_data->status_code != AWS_HTTP_STATUS_CODE_200_OK || user_data->current_result.len == 0) {
-        s_update_token_safely(user_data->client, NULL, false);
+        AWS_LOGF_DEBUG(
+            AWS_LS_IMDS_CLIENT,
+            "(id=%p) IMDS client failed to fetch token for requester %p, fall back to insecure path for the same "
+            "requester. Received response status code: %d",
+            (void *)user_data->client,
+            (void *)user_data,
+            user_data->status_code);
+        s_update_token_safely(user_data->client, NULL, false, 0 /*expire_timestamp*/);
     } else {
         struct aws_byte_cursor cursor = aws_byte_cursor_from_buf(&(user_data->current_result));
         aws_byte_cursor_trim_pred(&cursor, aws_char_is_space);
         aws_byte_buf_reset(&user_data->imds_token, true /*zero contents*/);
         if (aws_byte_buf_append_and_update(&user_data->imds_token, &cursor)) {
```

Comment:
I was looking through this code and was wondering if the `aws_byte_buf_append_and_update` can fail? We can probably just get rid of this if fails branch.

Reply:
I don't want to fix this, there are too many stuff in the codebase that we should fix...

I gave up on this rabbit hole and left todos for reference if we decide to revamp the client.